src/components/Modals/Send/utils.ts

```diff
@@ -151,7 +151,7 @@ export const handleSend = async ({
       if (!shouldUseEIP1559Fees && gasPrice === undefined) {
         throw new Error(`useFormSend: missing gasPrice for non-EIP-1559 tx`)
       }
-      const contractAddress = tokenOrUndefined(fromAssetId(asset.assetId).assetReference)
+      const contractAddress = contractAddressOrUndefined(asset.assetId)
```

Comment:
suggestion same comment re: rename to `maybeContractAddress` or do the `isToken` declarative dance